lib/hamlit/parser/haml_helpers.rb

```diff
@@ -111,7 +113,7 @@ def non_haml
     #   @yield The block within which to escape newlines
     def find_and_preserve(input = nil, tags = haml_buffer.options[:preserve], &block)
       return find_and_preserve(capture_haml(&block), input || tags) if block
-      tags = tags.each_with_object('') do |t, s|
+      tags = tags.each_with_object(+'') do |t, s|
```

Comment:
This file is _forked_ from Haml. The long-term idea is to carve out haml-parser.gem from haml.gem and share it between Haml and Hamlit, but it hasn't been done yet. So currently we're maintaining the files under `lib/hamlit/parser/*.rb` by basically copying Haml's files and replacing class names.

The original `lib/haml/helpers.rb` already supports `frozen_string_literal: true`, but in a different manner. To avoid further deviations, could you synchronize the implementation from https://github.com/haml/haml/blob/770962fec4cfe9ad2d9c6cc676b63d92932e7634/lib/haml/helpers.rb#L111-L119?